akka-projection-core/src/main/scala/akka/projection/StatusReporter.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.projection
+
+trait StatusReporter[-Envelope] {
```

Comment:
@jroper Would this work for the things you had in mind?
cc @hseeberger

Reply:
Thanks, @patriknw. Not exactly sure what is needed in detail, I'll leave that for James. From looking at the API I'd not call that "reporter", because it's not actively reporting, but instead gets called. So maybe `StatusCallback`?

Reply:
ok, renamed to `StatusObserver`